mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -205,9 +262,25 @@ class LogisticRegressionWithLBFGS
 
   override val optimizer = new LBFGS(new LogisticGradient, new SquaredL2Updater)
 
-  override protected val validators = List(DataValidators.binaryLabelValidator)
+  validators = List(DataValidators.binaryLabelValidator)
```

Comment:
Could we check whether the following works?

~~~
private val multiLabelValidator: (Double) => Boolean = { label =>
    (use numClasses from this instance)
}
~~~